doorstop/cli/main.py

```diff
@@ -180,8 +181,9 @@ def _edit(subs, shared):
                        help="edit document as exported XLSX")
     required = sub.add_argument_group('required arguments')
     required.add_argument('-T', '--tool', metavar='PROGRAM',
-                          help="text editor to open the document item",
-                          required=True)
+                          default=EDITOR,
+                          required=not bool(EDITOR),
```

Comment:
_Does_ this ever need to be required?

If I unset `$EDITOR` and run `$ doorstop reorder REQ` it will still open up the YAML file in my default text editor. I think we should have the same fallback behavior for editing items.

Reply:
On osx I get 
```
$ pipenv run doorstop edit REQ001
No application knows how to open /Users/aaronh/src/doorstop/reqs/REQ001.yml.
ERROR: no default editor for: /Users/aaronh/src/doorstop/reqs/REQ001.yml
```

Reply:
Ah, OK. I must have set a default editor for `.yml` on my system.

In that case, I think same `required=not bool(EDITOR)` logic should be added for `$ doorstop reorder`. I assume the reorder command fails on your machine when you unset `$EDITOR`?

Reply:
@jacebrowning No. Reorder simply goes through with auto-reordering.

```
export EDITOR=
pipenv run doorstop reorder REQ
building tree...
reorder from 'reqs/index.yml'? [y/n] y
reordering document REQ...
reordered document: REQ
```

I think your initial suggestion would be best, to simply remove `required` for the editor. The Operating System message about no application available to open the specific file is fine.

I have updated the PR to reflect this.